pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -520,3 +540,37 @@ func (cra *cpuResourceAdvisor) setPoolRegions(poolName string, regions []region.
 	}
 	return cra.metaCache.SetPoolInfo(poolName, pool)
 }
+
+// shareRegionRequirement refers to the resource demand of each share region,
+// sharePoolSize represents the total capacity of the share pool.
+// If the total resource demand of all share regions exceeds the total capacity of the pool,
+// we need to distribute the total capacity proportionally according to the resource requirement.
+func genShareRegionPools(shareRegionRequirement map[string]int, sharePoolSize int) map[string]int {
+	sharePools := make(map[string]int)
+
+	requirementSum := 0
+	for _, requirement := range shareRegionRequirement {
+		requirementSum += requirement
+	}
+	if requirementSum <= sharePoolSize {
+		return shareRegionRequirement
+	}
+
+	for poolName, requirement := range shareRegionRequirement {
+		sharePools[poolName] = sharePoolSize * requirement / requirementSum
```

Comment:
Better to use float64 for requirement and requirementSum to avoid loss of precision

Reply:
Poolsize is an integer, so it is reasonable to use int

Reply:
But requirement should be float64